Assets/MixedRealityToolkit.Tests/PlayModeTests/BaseCursorTests.cs

```diff
@@ -0,0 +1,339 @@
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+
+            // Left hand, pointer on cube, hand pinched
+            yield return rightHand.Hide();
+            yield return leftHand.Show(Vector3.zero);
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            VerifyCursorState(inputSystem.GazeProvider.GazeCursor, CursorStateEnum.Select);
+
+            // Both hands, pointer on cube, left pinched & right open
+            yield return rightHand.Show(Vector3.zero);
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            VerifyCursorState(inputSystem.GazeProvider.GazeCursor, CursorStateEnum.Select);
+
+            // Both hands, pointer on cube, both pinched
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            VerifyCursorState(inputSystem.GazeProvider.GazeCursor, CursorStateEnum.Select);
+
+            // Both hands, pointer on cube, right pinched & left open
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            VerifyCursorState(inputSystem.GazeProvider.GazeCursor, CursorStateEnum.Select);
+
+            // Restore the input simulation profile
+            iss.InputSimulationProfile = oldIsp;
+            yield return null;
+        }
+
+        [UnityTest]
+        public IEnumerator CursorContextMove()
+        {
+            var inputSystem = PlayModeTestUtilities.GetInputSystem();
+
+            // The cube needs to be moved from under the gaze cursor before we add the manipulation handler.
+            // Because the cube is under the gaze cursor from the beginning, it gets a focus gained event
+            // in Setup(). When we show the right hand, we get a focus lost event from the gaze pointer. 
+            // This messes with the CursorContextManipulationHandler hoverCount, as it decrements without
+            // ever having incremented. To avoid this, we move the cube out of focus before we add the
+            // ManipulationHandler and CursorContextManipulationHandler.
+            cube.transform.localPosition = new Vector3(0, -2, 2);
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            cube.AddComponent<ManipulationHandler>();
+            var temp = cube.AddComponent<CursorContextManipulationHandler>();
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            // Move cube back to original postion (described above)
+            cube.transform.localPosition = new Vector3(0, 0, 2);
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            // Show right hand on object
+            var rightHand = new TestHand(Handedness.Right);
+            Vector3 rightPos = new Vector3(0.05f, 0, 1.5f);
+            yield return rightHand.Show(rightPos);
+            yield return new WaitForFixedUpdate();
```

Comment:
I believe we don't need Wait for fixed update and null here, because hand.Show should wait for a frame for hand to show,

Reply:
In order to get the test to pass, I need `yield return null` for _Pinch right hand_, _Show left hand on object_ and _Pinch left hand_. I'm not sure why, but it won't work without them.